osmoutils/binary_search_test.go

```diff
@@ -41,11 +48,17 @@ func TestBinarySearch(t *testing.T) {
 }
 
 func TestBinarySearchNonlinear(t *testing.T) {
```

Comment:
Same here about the non-linear name

Reply:
What is the difference from `TestBinarySearch`? Why can't we have a single test `TestBinarySearch` that combines all of `TestBinarySearch`, `TestBinarySearchNonlinear`, `TestBinarySearchNonlinearNonzero`.

For each test case, we can have a name that describes whether it is a linear function and non-zero tolerance

Reply:
This is to avoid code duplication, each test fixture is essentially the same right now:
```
	for _, tc := range tests {
		actualSolvedInput, err := BinarySearch(tc.f, tc.lowerbound, tc.upperbound, tc.targetOutput, tc.errTolerance, tc.maxIterations)
		if tc.expectErr {
			require.Error(t, err)
		} else {
			require.NoError(t, err)
			require.True(sdk.IntEq(t, tc.expectedSolvedInput, actualSolvedInput))
		}
	}
```

Reply:
This is a great idea, I'll do that.